### PR TITLE
core: send version info to atlas

### PR DESCRIPTION
**Pull request by jbardin**
Set a default `Terraform-Version` header in atlas requests.

Also adds a helper function to build the version string, since we were formatting that in various places throughout the code.

### CODE REVIEW

**Comment by radeksimko** [comment]
I'm wondering if there's any way to incorporate this new helper function into the AWS provider too...
https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/config.go#L321-L327

Maybe the AWS SDK interface doesn't allow it, in which case feel free to ignore me... but it would be nice to keep this consistent.

**Comment by jbardin** [comment]
Yes, I looked at that, but I mistakingly looked at the function definition and thought it was in our code and required further refactoring. I think it's probably fine to use the same string there too. I'll update the PR.

File: builtin/providers/azurerm/config.go [hunk 1]
```diff
@@ -92,13 +92,7 @@ func withRequestLogging() autorest.SendDecorator {
...
 
 func setUserAgent(client *autorest.Client) {
-	var version string
-	if terraform.VersionPrerelease != "" {
-		version = fmt.Sprintf("%s-%s", terraform.Version, terraform.VersionPrerelease)
-	} else {
-		version = terraform.Version
-	}
-
+	version := terraform.VersionString()
```
Comment (stack72) [inline builtin/providers/azurerm/config.go]: Love that you added it to AzureRM as well - thanks!

**Comment by stack72** [comment]
LGTM! nice one :)

**Pull request merged**